### PR TITLE
Warn only for open files

### PR DESCRIPTION
**Pull request by rictic**
Align our warning behavior with other LSP providers. We should report warnings only for currently open files. When a user closes a file, clear out its warnings. When a user changes a file, refresh the list of warnings for all open files.

 - [x] CHANGELOG.md has been updated

### CODE REVIEW

File: src/polymer-language-server.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ connection.onInitialize((params): InitializeResult => {
     urlLoader: new FSUrlLoader(workspacePath),
     urlResolver: new PackageUrlResolver(), polymerJsonPath
   });
-  documents.all().forEach(d => scanDocument(d));
+  mapAllInitialDocuments(editorService);
```
Comment (aomarks) [inline src/polymer-language-server.ts]: Don't need to await here?

File: src/polymer-language-server.ts [hunk 2]
```diff
@@ -106,12 +106,46 @@ connection.onInitialize((params): InitializeResult => {
   };
 });
 
+async function mapAllInitialDocuments(editorService: EditorService) {
```
Comment (aomarks) [inline src/polymer-language-server.ts]: Why does this function take an editorService parameter, but other functions use the module scoped variable?

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,17 +10,17 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
...
 ## 1.3.3 - 2017-03-20
 
 - Update the linter to get a ton of new rules.
 - Configure the legacy interface (used by Atom) to read polymer.json, which will cause Atom to start seeing warnings from the linter.
 - A ton of bug fixes, especially around Polymer 2.0 elements.
 
-## Unreleased
-
-* Update dependencies.
-* Initialize the legacy protocol server with the expected path to the `polymer.json` file. This should allow the atom plugin (and possibly the sublime plugin) to get warnings from polymer-lint as well.
```
Comment (aomarks) [inline CHANGELOG.md]: Why were these bullets removed?

Comment (rictic) [inline CHANGELOG.md]: This was included here as a mistake. It was already fixed on master though.

File: src/polymer-language-server.ts (hunk 2 above)
Comment (rictic) [inline src/polymer-language-server.ts]: That way we can be sure that editorService is defined. Noticing that we have tons of checks for undefined checks for editorService was part of what inspired #62

File: src/polymer-language-server.ts (hunk 1 above)
Comment (rictic) [inline src/polymer-language-server.ts]: That's right. The containing function isn't async. Really we should wait until after returning. This too is handled much better in #62

**Comment by rictic** [comment]
Merging without Justin's review, as he's reviewed #62 which includes this change.

File: src/polymer-language-server.ts (hunk 2 above)
Comment (justinfagnani) [inline src/polymer-language-server.ts]: This name doesn't clearly mean something to me. Add docs or change name?

File: src/polymer-language-server.ts [hunk 4]
```diff
@@ -106,12 +106,46 @@ connection.onInitialize((params): InitializeResult => {
...
+
 // The content of a text document has changed. This event is emitted
 // when the text document is first opened or when its content has changed.
 documents.onDidChangeContent((change) => {
-  scanDocument(change.document, connection);
+  return handleErrors(handleChangedDocument(change.document), undefined);
 });
 
+async function handleChangedDocument(document: TextDocument) {
+  if (editorService) {
```
Comment (justinfagnani) [inline src/polymer-language-server.ts]: when is there not an `editorService`?

Comment (rictic) [inline src/polymer-language-server.ts]: This is why #62 manages the editor service instance properly. This was just paranoia on my part.

File: src/polymer-language-server.ts (hunk 2 above)
Comment (rictic) [inline src/polymer-language-server.ts]: Name changed in #62 (and it turns out this function isn't needed at all, you get events for each open document)

**Comment by rictic** [comment]
Sorry for merging while you were reviewing Justin!

**Pull request merged**